examples/Enzyme/PowerElectronics/main.cpp

```diff
@@ -0,0 +1,183 @@
+    constexpr double Kpv_ = 0.1;
+    constexpr double Kic_ = 2.0e4;
+    constexpr double Kpc_ = 15.0;
+    constexpr double Cf_ = 5.0e-5;
+    constexpr double rLf_ = 0.1;
+    constexpr double Lf_ = 1.35e-3;
+    constexpr double rLc_ = 0.03;
+    constexpr double Lc_ = 0.35e-3;
+
+    constexpr bool ref_frame_ = true;
+
+    std::vector<double> yp_(0);
+
+    double omega = wb_ - mp_ * y_[4];
+    if (ref_frame_)
+    {
+        f_[0] = omega - y_[0];
+    }
+    else
+    {
+        f_[0] = 0.0;
+    }
+    
+    f_[1] = cos(y_[3]) * y_[14] - sin(y_[3]) * y_[15];
+    f_[2] = sin(y_[3]) * y_[14] + cos(y_[3]) * y_[15];
+
+    double vbd_in = cos(y_[3]) * y_[1] + sin(y_[3]) * y_[2];
+    double vbq_in = -sin(y_[3]) * y_[1] + cos(y_[3]) * y_[2];
+
+    f_[3] = -yp_[3] + omega - y_[0];
+    f_[4] = -yp_[4] + wc_ * (y_[12] * y_[14] + y_[13] * y_[15] - y_[4]);
+    f_[5] = -yp_[5] + wc_ * (-y_[12] * y_[15] + y_[13] * y_[14] - y_[5]);
+
+    double vod_star = Vn_ - nq_ * y_[5];
+    double voq_star = 0.0;
+
+    f_[6] = -yp_[6] + vod_star - y_[12];
+    f_[7] = -yp_[7] + voq_star - y_[13];
+
+    double ild_star = F_ * y_[14] - wb_ * Cf_ * y_[13] + Kpv_ * (vod_star - y_[12]) + Kiv_ * y_[6];
+    double ilq_star = F_ * y_[15] + wb_ * Cf_ * y_[12] + Kpv_ * (voq_star - y_[13]) + Kiv_ * y_[7];
+
+    f_[8] = -yp_[8] + ild_star - y_[10];
+    f_[9] = -yp_[9] + ilq_star - y_[11];
+
+    double vid_star = -wb_ * Lf_ * y_[11] + Kpc_ * (ild_star - y_[10]) + Kic_ * y_[8];
+    double viq_star = wb_ * Lf_ * y_[10] + Kpc_ * (ilq_star - y_[11]) + Kic_ * y_[9]; 
+    
+    f_[10] = -yp_[10] - (rLf_ / Lf_) * y_[10] + omega * y_[11] +  (vid_star - y_[12]) / Lf_;
+    f_[11] = -yp_[11] - (rLf_ / Lf_) * y_[11] - omega * y_[10] +  (viq_star - y_[13]) / Lf_;
+
+    f_[12] = -yp_[12] + omega * y_[13] + (y_[10] - y_[14]) / Cf_;
+    f_[13] = -yp_[13] - omega * y_[12] + (y_[11] - y_[15]) / Cf_;
+
+    f_[14] = -yp_[14] - (rLc_ / Lc_) * y_[14] + omega * y_[15] + (y_[12] - vbd_in) / Lc_;
+    f_[15] = -yp_[15] - (rLc_ / Lc_) * y_[15] - omega * y_[14] + (y_[13] - vbq_in) / Lc_;
+}
+
+template <typename T>
+void EnzymeModelJacobian(T* model, DenseMatrix& jac) {
+    int N = model->size();
+    std::vector<double> y(N);
+    std::vector<double> v(N);
+    std::vector<double> res(N);
+    std::vector<double> d_res(N);
+    for (int idy = 0; idy < N; ++idy)
+    {
+        // Elementary vector for Jacobian-vector product
+        for (int idx = 0; idx < N; ++idx)
+        {
+            y[idx] = (model->y())[idx];
+            res[idx] = (model->getResidual())[idx];
+            v[idx] = 0.0;
+        }
+        v[idy] = 1.0;
+  
+        // Autodiff
+        __enzyme_fwddiff((void*)evaluateResidual, 
+                         enzyme_dup, y, v,
+                         enzyme_dupnoneed, res, &d_res);
+  
+        // Store result
+        for (int idx = 0; idx < N; ++idx)
+        {
+            jac.setValue(idx, idy, d_res[idx]);
+        }
+    }
+}
+
```

Comment:
Add comment that this code check auto diff of the Distributed Generator function $\frac{d}{dy} F(yp,y)$. Its not too obvious unless reading the code.

Reply:
Addressed in [495ab8c](https://github.com/ORNL/GridKit/pull/54/commits/495ab8c1f7f0865fd911f25d2e9bc1d819273223)